payment/src/main/java/org/killbill/billing/payment/bus/PaymentBusEventHandler.java

```diff
@@ -34,42 +34,112 @@
+
+                final Account account;
+                try {
+
+                    account = accountApi.getAccountById(event.getAccountId(), internalContext);
+
+                    invoicePaymentInternalApi.createPurchaseForInvoicePayment(false,
+                                                                              account,
+                                                                              event.getInvoiceId(),
+                                                                              account.getPaymentMethodId(),
+                                                                              null,
+                                                                              amountToBePaid,
+                                                                              account.getCurrency(),
+                                                                              null,
+                                                                              null,
+                                                                              null,
+                                                                              Collections.emptyList(),
+                                                                              new PaymentOptions() {
+                                                                                  @Override
+                                                                                  public boolean isExternalPayment() {
+                                                                                      return false;
+                                                                                  }
+
+                                                                                  @Override
+                                                                                  public List<String> getPaymentControlPluginNames() {
+                                                                                      return paymentControlPluginNames;
+                                                                                  }
+                                                                              },
+                                                                              internalContext);
+                } catch (final AccountApiException e) {
+                    log.warn("Failed to process invoice payment", e);
+                } catch (final PaymentApiException e) {
+                    // Look for a LockFailedException cause to decide if we should reschedule
+                    // (It should be the first cause of the PaymentApiException, but we support additional levels mostly for testing)
+                    Throwable curCause = e.getCause();
+                    while (curCause != null) {
+                        if (curCause instanceof LockFailedException) {
+                            throw new QueueRetryException();
```

Comment:
Should we make the retry schedule configurable (shared config for all of these), instead of relying on `DEFAULT_RETRY_SCHEDULE`? To your point that we may eventually timeout, we probably want to have a default with more retries (e.g. `1m,1m,1m,5m,5m,5m,1h,1h,1h,6h,6h,6h,24h,24h,24h`).

Reply:
Updated the implementation with per module config as discussed: https://github.com/killbill/killbill/pull/1888/commits/b2f74ab2a35e0145f64a5f2275c0a629ce3ab96f

Reply:
> providing an empty list does not seem handled

Yup, agreed that empty list should result in no retry. That being said, I don't see why we would want to disable these specific retries.